Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Systems/PortalArtifactSystem.cs

```diff
@@ -21,22 +36,52 @@ public override void Initialize()
     private void OnActivate(Entity<PortalArtifactComponent> artifact, ref ArtifactActivatedEvent args)
     {
         var map = Transform(artifact).MapID;
-        var firstPortal = Spawn(artifact.Comp.PortalProto, _transform.GetMapCoordinates(artifact));
+        var validMinds = new List<EntityUid>();
+        var mindMobs = new HashSet<Entity<MindContainerComponent, TagComponent>>();
 
-        var minds = new List<EntityUid>();
-        var mindQuery = EntityQueryEnumerator<MindContainerComponent, TransformComponent>();
-        while (mindQuery.MoveNext(out var uid, out var mc, out var xform))
+        _lookup.GetEntitiesOnMap<MindContainerComponent, TagComponent>(map, mindMobs);
+        foreach (var comp in mindMobs)
         {
-            if (mc.HasMind && xform.MapID == map)
-                minds.Add(uid);
```

Comment:
Just add a isentityincontainer check here and remove all the other code.

Reply:
thanks for the hint. This new version is SO much better now.